File: server/api/src/view_models/user_view_model.rs
use serde::Serialize;
use uuid::Uuid;

use super::{
    assets::base_models::asset::IdentifiableAssetViewModel,
    portfolio_account_view_model::PortfolioAccountViewModel,
};

#[derive(Debug, Serialize)]
pub struct UserViewModel {
    pub id: Uuid,
    pub username: String,
    pub default_asset_id: IdentifiableAssetViewModel,
    pub portfolio_accounts: Vec<PortfolioAccountViewModel>,
    pub custom_assets: Vec<IdentifiableAssetViewModel>,
}

File: web/src/models/userViewModel.ts
import { PortfolioAccountViewModel } from "./portfolioAccountViewModel";
import { AssetViewModel } from "./assetViewModel";

export interface UserViewModel {
  id: string;
  username: string;
  default_asset_id: AssetViewModel;
  portfolio_accounts: PortfolioAccountViewModel[];
  custom_assets: AssetViewModel[];
}
